k3k-kubelet/provider/node.go

```diff
@@ -20,3 +23,72 @@ func (n *Node) Ping(context.Context) error {
 func (n *Node) NotifyNodeStatus(ctx context.Context, cb func(*corev1.Node)) {
 	n.notifyCallback = cb
 }
+
+// nodeConditions returns the basic conditions which mark the node as ready
+func nodeConditions() []v1.NodeCondition {
+	return []v1.NodeCondition{
+		{
+			Type:               "Ready",
+			Status:             v1.ConditionTrue,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletReady",
+			Message:            "kubelet is ready.",
+		},
+		{
+			Type:               "OutOfDisk",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletHasSufficientDisk",
+			Message:            "kubelet has sufficient disk space available",
+		},
+		{
+			Type:               "MemoryPressure",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletHasSufficientMemory",
+			Message:            "kubelet has sufficient memory available",
+		},
+		{
+			Type:               "DiskPressure",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "KubeletHasNoDiskPressure",
+			Message:            "kubelet has no disk pressure",
+		},
+		{
+			Type:               "NetworkUnavailable",
+			Status:             v1.ConditionFalse,
+			LastHeartbeatTime:  metav1.Now(),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "RouteCreated",
+			Message:            "RouteController created a route",
+		},
+	}
+}
+
+func ConfigureNode(node *v1.Node, hostname string, servicePort int, ip string) {
+	node.Status.Conditions = nodeConditions()
+	node.Status.DaemonEndpoints.KubeletEndpoint.Port = int32(servicePort)
+	node.Status.Addresses = []v1.NodeAddress{
+		{
+			Type:    v1.NodeHostName,
+			Address: hostname,
+		},
+		{
+			Type:    v1.NodeInternalIP,
+			Address: ip,
+		},
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourceCPU:    resource.MustParse("8"),
+		v1.ResourceMemory: resource.MustParse("326350752922"),
+		v1.ResourcePods:   resource.MustParse("110"),
+	}
```

Comment:
Are this random? Maybe we should add a comment if we need to change them somehow.

Reply:
This is copied from where it originally was in the code base. I didn't make and changes or decisions on the values.